api.planx.uk/admin/migrations/splitFindProperty.ts

```diff
@@ -0,0 +1,165 @@
+import { Request, Response, NextFunction } from 'express';
+import { gql } from "graphql-request";
+import { customAlphabet } from "nanoid-good";
+import en from "nanoid-good/locale/en";
+
+import { adminGraphQLClient as client } from "../../hasura";
+import { Flow, PublishedFlow } from "../../types";
+
+/**
+ * This migration ran in February 2023
+ * 
+ * Context: FindProperty is split from two pages into two separate components (FindProperty + PlanningInformation) to faciliate overriding the "property.type"
+ * 
+ * What this does: Inserts a PlanningInformation component with default props (override = false) directly after a FindProperty in a flow schema. Additionally updates the currently published version of that flow
+ * 
+ * Considerations:
+ *   - nodeIds for the new PlanningInformation must be unique among flows to prevent clone behavior if used as portals
+ *   - a flow and its' currently published version need to share a nodeId so that it's not picked up in the publishing diff
+ *   - we assume one FindProperty per flow
+ *   - we assume no flow has PropertyInformation yet
+ *   - we ignore external portal nodes when scanning a flow schema (external portals are flows and can be migrated independently)
+ *   - we do not insert a corresponding operation, as "migrations" don't have actors, shouldn't need to be rolled-back, etc
+ * 
+ */
+
+const findPropertyType = 9;
+
+const defaultPropertyInformationNode: Record<"type" | "data", any> = {
+  "type": 12,
+  "data": {
+    "title": "About the property",
+    "description": "This is the information we currently have about the property",
+    "showPropertyTypeOverride": false
+  }
+};
+
+// aligns with editor.planx.uk/src/@planx/graph
+const uniqueId = customAlphabet(en)(
+  "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789",
+  10
+);
```

Comment:
i'm not fully caught up on `planx-client` work yet, but I think this may be a good future candidate?? it's repeated a couple times in the editor, and now I've added these dependencies to the API - which feels a like a lot of added overhead for a migration we'll run once, but node id generation feels important to do consistently.